feat: New Coverage engine status checks - post-release tweaks ALA-695 (#1958)
codacy/docs: docs/release-notes/cloud/cloud-2023-11-23-new-coverage-engine-status-checks.md

# New Coverage engine status checks November 23rd, 2023

> !!! info "This release note applies if you have set Codacy to send pull request status data to your Git provider (see how on [GitHub](../../repositories-configure/integrations/github-integration.md#status-checks), [GitLab](../../repositories-configure/integrations/gitlab-integration.md#pull-request-status), and [Bitbucket](../../repositories-configure/integrations/bitbucket-integration.md#pull-request-status))."<!--NOTE Temporary comment for review purpose, to remove-->

Author: Small tweak for clarity:

```suggestion
!!! info "This release note applies to the repositories for which you set Codacy to send pull request coverage status data to your Git provider (see how on [GitHub](../../repositories-configure/integrations/github-integration.md#status-checks), [GitLab](../../repositories-configure/integrations/gitlab-integration.md#pull-request-status), and [Bitbucket](../../repositories-configure/integrations/bitbucket-integration.md#pull-request-status))."
```

On November 23rd 2023 we activated the new, faster Coverage engine and set it to send coverage data to your Git provider. This is part of an ongoing effort to improve the speed and value of the insights provided by Codacy.

As a consequence of this update, you can now see two additional checks on your pull requests, marked **beta**, alongside the preexisting Codacy Coverage check. If you are using the Codacy Coverage status check to block merging pull requests on GitHub, please [update your setup](#if-you-are-using-the-old-coverage-status-check-to-block-merging-pull-requests-on-github) and avoid disruptions in the future.<!--NOTE Temporary comment for review purpose, to remove-->

This is what the new checks look like on each provider:

-   On GitHub:

>     ![New Coverage status checks GitHub](../images/ala-695-status-checks-github.png)<!--NOTE Temporary comment for review purpose, to remove-->

Author: Although it would be nice, it's a minor improvement requiring a lot of effort to recreate the screenshots (especially on GL and BB). I suggest skipping this one 🤔

## Deprecation and removal of the old Coverage engine status checks

> Currently, Codacy sends coverage data to your Git provider from both the old and new Coverage engines. We will deprecate and eventually remove the old status checks on the following dates:<!--NOTE Temporary comment for review purpose, to remove-->

Author: Maybe "later" conveys confidence?
```suggestion
Currently, Codacy sends coverage data to your Git provider from both the old and new Coverage engines. We will deprecate and later remove the old status checks on the following dates:
```

> Currently, Codacy sends coverage data to your Git provider from both the old and new Coverage engines. We will deprecate and eventually remove the old status checks on the following dates:<!--NOTE Temporary comment for review purpose, to remove-->

Reviewer: Yes, "later" works fine! 👍 
I was unsure if the "eventually" was intended... 😉

      <td>February 14th, 2024</td>
      <td>Deprecation of the old Coverage status check</td>
      <td>

>         <p>From this day on, the new checks will no longer be marked <strong>beta</strong> and will be the official status checks for Codacy Coverage. The old check will be marked <strong>deprecated</strong> but will keep working.</p><!--NOTE Temporary comment for review purpose, to remove-->

Reviewer: I would switch these around (order by importance):

```suggestion
        <p>From this day on, the new checks will be the official status checks for Codacy Coverage and will no longer be marked <strong>beta</strong>. The old check will be marked <strong>deprecated</strong> but will keep working.</p><!--NOTE Temporary comment for review purpose, to remove-->
```

        <p>If on this date you are still using the old check to block merging pull requests on GitHub, please <a href="#if-you-are-using-the-old-coverage-status-check-to-block-merging-pull-requests-on-github">update your Git provider settings</a>.</p>
      </td>
    </tr>
    <tr>
      <td>April 18th, 2024</td>
      <td>Brownout of the old Coverage status check</td>
      <td>
        <p>Codacy will not send the old status check on this day.</p><!--NOTE Temporary comment for review purpose, to remove-->

>         <p>This will help you confirm that your setup has been updated correctly and lower the risk of disruptions when the old status check is removed.</p>

Reviewer: How? What are the steps to confirm if the setup has been updated correctly or not?

Author: How about this? Unfortunately, I don't have more details about "how" it would break. It's just meant to break for one day if the new setup is not correct.

```suggestion
        <p>If, on this day, you can merge your pull requests using the new checks and don't encounter other issues, it means that your setup has been updated correctly.</p>
```

Reviewer: I would keep the sentence describing the goal and would add how can they confirm.
WDYT?

```suggestion
        <p>This will help you confirm that your setup has been updated correctly and lower the risk of disruptions when the old status check is removed.<br/>If on this day you can merge your pull requests using the new checks and don't encounter other issues, it means that your setup has been updated correctly. Otherwise, if on this date you are still using the old check to block merging pull requests on GitHub, please <a href="#if-you-are-using-the-old-coverage-status-check-to-block-merging-pull-requests-on-github">update your Git provider settings</a>.</p>
```

Author: LGTM 👍

> ## If you are using the old Coverage status check to block merging pull requests on GitHub

Author: That's correct, see [this thread](https://codacy.slack.com/archives/C04EJUDAL8J/p1700739963069869?thread_ts=1700647863.983539&cid=C04EJUDAL8J) 👀 

I was surprised too...

> ## If you are using the old Coverage status check to block merging pull requests on GitHub

Reviewer: O.....k.... 😕

> If you are using the old status check to block merging pull requests on GitHub, [remove the old Codacy Coverage report as a required check by editing the branch protection rules](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule#editing-a-branch-protection-rule) of your pull requests' target branch under the section **Require status checks to pass before merging**:

Reviewer: Splitting into two steps, for clarity.

```suggestion
If you are using the old status check to block merging pull requests on GitHub, remove the old Codacy Coverage report as a required check of your pull requests' target branch:

1.  [Edit the branch protection rules](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule#editing-a-branch-protection-rule) of your pull requests' target branch.
1.  Remove the old **Codacy Coverage Report** under the section **Require status checks to pass before merging**:
```

Author: I'm not sure we should split here, since by "removing the old Codacy Coverage Report" status check, the user is "editing the branch protection rules", so these two steps are really one action.

At most, the first step should be "Navigate to the branch protection rules page" or similar. Thoughts?

![New Coverage status checks GitHub](../images/ala-695-update-status-checks-github.png)

> ## Differences in coverage metrics between the old and new Coverage enginesx

Reviewer: Fixing my typo. 🙏 

```suggestion
## Differences in coverage metrics between the old and new Coverage engines
```

> You may notice some differences in the coverage metrics reported by the old and new Coverage engines. This may happen because the new Coverage engine calculates coverage metrics considering all the files included in the coverage report, while the old Coverage engine ignores some files:<!--NOTE Temporary comment for review purpose, to remove-->

Author: > Will customers need to adjust their quality gates for coverage?

They may have to, but we don't know how the numbers will change since each case will differ. The metrics may increase, decrease, or stay the same. The best we can do is mention that they may have to update their rules.

```suggestion
You may notice some differences in the coverage metrics reported by the old and new Coverage engines and may need to update your [coverage gate rules](../../repositories-configure/adjusting-quality-gates.md) accordingly. This may happen because the new Coverage engine calculates coverage metrics considering all the files included in the coverage report, while the old Coverage engine ignores some files:
```

## If you are using the old Coverage status check to block merging pull requests on GitHub

If you are using the old status check to block merging pull requests on GitHub, remove the old Codacy Coverage Report as a required check of your pull requests' target branch:

> 1.  [Edit the the branch protection rule](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule#editing-a-branch-protection-rule) of your pull requests' target branch.

Reviewer: ```suggestion
1.  [Edit the branch protection rule](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule#editing-a-branch-protection-rule) of your pull requests' target branch.
```
